src/System.Text.Json/src/System/Text/Json/Reader/Utf8JsonReader.TryGet.cs

```diff
@@ -470,5 +493,56 @@ public bool TryGetDateTimeOffset(out DateTimeOffset value)
             ReadOnlySpan<byte> span = HasValueSequence ? ValueSequence.ToArray() : ValueSpan;
             return JsonHelpers.TryParseAsISO(span, out value, out int bytesConsumed) && span.Length == bytesConsumed;
         }
+
+        /// <summary>
+        /// Reads the next JSON token value from the source and parses it to a <see cref="Guid"/>.
+        /// Returns true if the entire UTF-8 encoded token value can be successfully
+        /// parsed to a <see cref="Guid"/> value.
+        /// Returns false otherwise.
+        /// </summary>
+        /// <exception cref="InvalidOperationException">
+        /// Thrown if trying to get the value of a JSON token that is not a <see cref="JsonTokenType.String"/>.
+        /// <seealso cref="TokenType" />
+        /// </exception>
+        public bool TryGetGuid(out Guid value)
+        {
+            if (TokenType != JsonTokenType.String)
+            {
+                throw ThrowHelper.GetInvalidOperationException_ExpectedString(TokenType);
+            }
+
+            ReadOnlySpan<byte> span = HasValueSequence ? ValueSequence.ToArray() : ValueSpan;
+            int bytesConsumed = 0;
+
+            switch (span.Length)
+            {
+                case 36:
+                    // Format D (default): nnnnnnnn-nnnn-nnnn-nnnn-nnnnnnnnnnnn
+                    return Utf8Parser.TryParse(span, out value, out bytesConsumed, 'D') && span.Length == bytesConsumed;
```

Comment:
I think we should **only** support the default GUID format for now (don't specify a standard format) rather than trying to guess and support all that format specifiers that `Utf8Parser` accepts. 

For example, we don't support 'X'.

I would only add custom support if there is user request for it, especially given @JamesNK previous feedback:
> I don't think methods for Guid are that important.

Reply:
@Tornhoof, since you asked for it initially, what are your thoughts on it?

Reply:
Yes, 'D' is great, all the talk I saw in json-schema about it also assumed the 'D' format

Reply:
Based on feedback, only 'D' will be supported.